firmware/ventilator-controller-stm32/CMakeLists.txt

```diff
@@ -38,15 +38,44 @@ include_directories(
 )
 
 if ("${CMAKE_BUILD_TYPE}" STREQUAL "TestCatch2")
+    set(CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/)
+    include(CodeCoverage)
+    append_coverage_compiler_flags()
+
+    # disable all optimization
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O0")
+    set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -O0")
+    set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -fno-inline -fno-inline-small-functions -fno-default-inline")
```

Comment:
How important are these flags? Are they needed for code coverage? If they're only needed for code coverage, how easy would it be to make a separate build target for running tests vs. for checking test coverage?